distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -1831,28 +1832,24 @@ type Table
             Table.Value self.name self.connection internal_columns ctx
 
     ## PRIVATE
-
-       Inserts a new row to the table.
-
-       Arguments:
-       - values: The values making up the row of the table.
-
-       It actually modifies the underlying table in the database.  It can only
-       be called on the Table if no operations modifying it have been performed
-       like modifying, removing or adding columns, filtering, grouping etc.
-    insert : Vector Any -> Nothing
-    insert self values =
-        table_name = case self.context.from_spec of
-            From_Spec.Table name _ -> name
-            _ -> Error.throw <| Illegal_State.Error "Inserting can only be performed on tables as returned by `query`, any further processing is not allowed."
-        # TODO [RW] before removing the PRIVATE tag, add a check that no bad stuff was done to the table as described above
-        pairs = self.internal_columns.zip values col-> value->
-            [col.name, SQL_Expression.Constant value]
-        query = self.connection.dialect.generate_sql <| Query.Insert table_name pairs
-        affected_rows = self.connection.execute_update query
-        case affected_rows == 1 of
-            False -> Error.throw <| Illegal_State.Error "The update unexpectedly affected "+affected_rows.to_text+" rows."
-            True -> Nothing
+       Checks if this table is a 'trivial query'.
+
+       A trivial query is a result of `connection.query` that has not been
+       further processed. If there are any columns that are added or removed, or
+       any other operations like join or aggregate are performed, the resulting
+       table is no longer trivial.
+
+       Some operations, like writing to tables, require their target to be a
+       trivial query.
+    is_trivial_query : Boolean ! Table_Not_Found
```

Comment:
1. Why would it be the context? it's a property of the whole table, exactly because of (2) - I also want to check if the columns are intact and unmodified

For example, `table.set "[X] + 2" "X"` will have context completely unchanged, but the contents of the column X will all be shifted by 2. And so when inserting to this table, should we join along values of the original X or the updated X? It is ill-defined.

2. And that's why - I don't think we should allow _any_ column modifications, be it rename or removal. After such operations it's no longer the same table, and we are inserting to the original one, not the modified one.

e.g. what if I have table T with columns A, B and C and remove C. Then I append to this new table a table with columns A and B and have `error_on_missing_columns=True`. `C` is missing from the table I will _actually_ append into, but not from the table that I set as my target. Do I error on this missing C or not? It's unclear.

So due to these examples, IMO it only makes sense to append to a 'trivial' table.